app/src/main/java/com/startup/go4lunch/api/OverpassRestaurantApi.java

```diff
@@ -0,0 +1,87 @@
+package com.startup.go4lunch.api;
+
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.startup.go4lunch.api.utilApi.OverpassCalls;
+import com.startup.go4lunch.api.utilApi.OverpassElements;
+import com.startup.go4lunch.api.utilApi.OverpassGsonObject;
+import com.startup.go4lunch.model.Restaurant;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class OverpassRestaurantApi implements RestaurantApi, OverpassCalls.Callback {
+
+    MutableLiveData<List<Restaurant>> restaurantListLiveData = new MutableLiveData<>();
+
+    public OverpassRestaurantApi() {
+        OverpassCalls.fetchRestaurantList(this);
+    }
+
+
+    @Override
+    public LiveData<List<Restaurant>> getRestaurantListLiveData() {
+        return this.restaurantListLiveData;
+    }
+
+    @Override
+    public void onResponse(@Nullable OverpassGsonObject overpassGsonObject) {
+        if (overpassGsonObject != null) {
+            this.restaurantListLiveData.setValue(overpassGsonObjectToRestaurantList(overpassGsonObject));
+        }
+    }
+
+    @Override
+    public void onFailure() {
+
+    }
+
+    private List<Restaurant> overpassGsonObjectToRestaurantList(OverpassGsonObject overpassGsonObject) {
+
+        List<Restaurant> restaurantList = new ArrayList<>();
+        List<OverpassElements> overpassElementslist = overpassGsonObject.getOverpassElementslist();
+
+        for (int i=0;i<overpassElementslist.size();i++) {
```

Comment:
Un pattern qui se retrouve souvent est de déplacer cette transformation d'un modèle de couche réseau dans une méthode static du modèle de la couche "business logic"
 
`static Restaurant from(OverpassElement element);` déclarée dans la classe Restaurant.


Et ici tu ne ferais qu'une simple boucle sur overpassElementslist

```java
for (element in overpassElementslist) {
    restaurantList.add(Restaurant.from(element))
}
```

Reply:
Aussi, si renvoyé par l'api overpass, vérifier que l'élément est bien un restaurant avant de créer un objet de type restaurant. La méthode pourrait renvoyer null sinon, et il faudrait filtrer dans la boucle.

Reply:
for (element in overpassElementslist) , je croit que c'est du Kotlin le "in" :)

Reply:
Oups, en effet. La syntaxe équivalente en java (plus pratique que les indexes quand tu parcours tous les éléments d'une liste) est 
`for (OverpassElement element : overpassElementList{...})`

Reply:
Ok